homeassistant/helpers/typing.py

```diff
@@ -16,3 +17,12 @@
 
 # Custom type for recorder Queries
 QueryType = Any
+
+
+class UndefinedType(Enum):
```

Comment:
Couldn't this be put in homeassistant.const or similar to avoid the circular dependency?

Reply:
Yes, please note that I already wrote the last paragraph of the original description about this very thing along with some considerations and history why it's not (yet?) done like that.

Reply:
Sorry, read the description and didn't directly look at the code